Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -821,7 +826,28 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
       }
     }
 
-    return .init(newItems)
+    /// Returns an leading trivia for the member blocks closing brace.
+    /// It will add a leading newline, if there is none.
+    var leadingTriviaForClosingBrace: Trivia {
+      guard !node.members.isEmpty || !newItems.isEmpty else {
+        return node.rightBrace.leadingTrivia
+      }
+
+      guard !node.rightBrace.leadingTrivia.contains(where: { $0.isNewline }) else {
+        return node.rightBrace.leadingTrivia
+      }
```

Comment:
I think these would read easier as `if` statements because we could avoid the negations then.